sdk/keyvault/keyvault/src/keys/index.ts

```diff
@@ -0,0 +1,478 @@
+  }
+
+  /**
+   * The Get Deleted Key operation returns the specified deleted key along with its attributes.
+   * This operation requires the keys/get permission.
+   * @summary Gets the specified deleted key.
+   * @param keyName The name of the key.
+   * @param [options] The optional parameters
+   * @returns Promise<DeletedKey>
+   */
+  public async getDeletedKey(
+    keyName: string,
+    options?: RequestOptions
+  ): Promise<DeletedKey> {
+    const response = await this.client.getDeletedKey(this.vaultBaseUrl, keyName, options ? options.requestOptions : {});
+    return this.getKeyFromKeyBundle(response);
+  }
+
+  /**
+   * The purge deleted key operation removes the key permanently, without the possibility of
+   * recovery. This operation can only be enabled on a soft-delete enabled vault. This operation
+   * requires the keys/purge permission.
+   * @summary Permanently deletes the specified key.
+   * @param keyName The name of the key.
+   * @param [options] The optional parameters
+   * @returns Promise<void>
+   */
+  public async purgeDeletedKey(keyName: string, options?: RequestOptions): Promise<void> {
+    await this.client.purgeDeletedKey(this.vaultBaseUrl, keyName, options ? options.requestOptions : {});
+  }
+
+  /**
+   * Recovers the deleted key in the specified vault. This operation can only be performed on a
+   * soft-delete enabled vault. This operation requires the keys/recover permission.
+   * @summary Recovers the deleted key to the latest version.
+   * @param keyName The name of the deleted key.
+   * @param [options] The optional parameters
+   * @returns Promise<Key>
+   */
+  public async recoverDeletedKey(
+    keyName: string,
+    options?: RequestOptions
+  ): Promise<Key> {
+    const response = await this.client.recoverDeletedKey(this.vaultBaseUrl, keyName, options ? options.requestOptions : {});
+    return this.getKeyFromKeyBundle(response);
+  }
+
+  /**
+   * Requests that a backup of the specified key be downloaded to the client. All versions of the
+   * key will be downloaded. This operation requires the keys/backup permission.
+   * @summary Backs up the specified key.
+   * @param keyName The name of the key.
+   * @param [options] The optional parameters
+   * @returns Promise<Uint8Array | undefined>
+   */
+  public async backupKey(
+    keyName: string,
+    options?: RequestOptions
+  ): Promise<Uint8Array | undefined> {
+    const response = await this.client.backupKey(this.vaultBaseUrl, keyName, options ? options.requestOptions : {});
+    return response.value;
+  }
+
+  /**
+   * Restores a backed up key, and all its versions, to a vault. This operation requires the
+   * keys/restore permission.
+   * @summary Restores a backed up key to a vault.
+   * @param keyBundleBackup The backup blob associated with a key bundle.
+   * @param [options] The optional parameters
+   * @returns Promise<Key>
+   */
+  public async restoreKey(
+    keyBundleBackup: Uint8Array,
```

Comment:
>keyBundleBackup [](start = 4, length = 15)

this parameter should be called simply backup

Reply:
Done